test/hybrid_vpn_test.go

```diff
@@ -0,0 +1,235 @@
+		terraform.Destroy(t, awsHybridVpnGcpTerraformOptions)
+	})
+
+	test_structure.RunTestStage(t, "create_aws_hybrid_vpn_gcp", func() {
+		terraform.InitAndApply(t, awsHybridVpnGcpTerraformOptions)
+	})
+
+	//
+	//
+	// Connectivity Validation
+	//
+	//
+
+	validationResourcesTerraformOptions := &terraform.Options{}
+	validationResourcesDir := test_structure.CopyTerraformFolderToTemp(t, ".", "modules/validation-resources")
+
+	test_structure.RunTestStage(t, "create_validation_resources_options", func() {
+		validationResourcesInputs := map[string]interface{}{
+			"aws_region":          awsRegion,
+			"aws_assume_role_arn": awsAssumeRoleArn,
+			"google_region":       gcpRegion,
+			"google_project":      gcpProject,
+			"google_credentials":  googleCredentials,
+			"name":                runId,
+			"aws_vpc_id":          terraform.Output(t, awsVpcBootstrapTerraformOptions, "vpc_id"),
+			"aws_subnet_id":       terraform.OutputList(t, awsVpcBootstrapTerraformOptions, "private_subnets")[0],
+			"gcp_network_name":    terraform.Output(t, gcpVpcBootstrapTerraformOptions, "network_name"),
+			"gcp_subnetwork_name": terraform.Output(t, gcpVpcBootstrapTerraformOptions, "primary_subnet_name"),
+			"gcp_zone_name":       gcpRegion + "-a",
+		}
+
+		validationResourcesTerraformOptions = terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+			TerraformDir: validationResourcesDir,
+			Vars:         validationResourcesInputs,
+		})
+
+		_ = copySupportingFiles([]string{"provider-aws.tf", "provider-gcp.tf", "extra-inputs.tf"}, validationResourcesDir)
+	})
+
+	defer test_structure.RunTestStage(t, "cleanup_validation_resources_support_files", func() {
+		_ = cleanupSupportingFiles([]string{"provider-aws.tf", "provider-gcp.tf", "extra-inputs.tf"}, validationResourcesDir)
+	})
+
+	defer test_structure.RunTestStage(t, "cleanup_validation_resources", func() {
+		terraform.Destroy(t, validationResourcesTerraformOptions)
+	})
+
+	test_structure.RunTestStage(t, "create_validation_resources", func() {
+		terraform.InitAndApply(t, validationResourcesTerraformOptions)
+	})
+
+	test_structure.RunTestStage(t, "validate_connectivity", func() {
+		sshKeyPair := ssh.KeyPair{
+			PrivateKey: terraform.Output(t, validationResourcesTerraformOptions, "ssh_private_key"),
+		}
+
+		sshAgent := ssh.SshAgentWithKeyPair(t, &sshKeyPair)
+		fmt.Println(sshKeyPair.PrivateKey)
+		defer sshAgent.Stop()
+
+		awsPrivateIp := terraform.Output(t, validationResourcesTerraformOptions, "ec2_instance_private_ip")
+		//awsPublicIp := terraform.Output(t, validationResourcesTerraformOptions, "ec2_instance_public_ip")
+		//gcpPrivateIp := terraform.Output(t, validationResourcesTerraformOptions, "gce_instance_private_ip")
+		gcpPublicIp := terraform.Output(t, validationResourcesTerraformOptions, "gce_instance_public_ip")
+
+		expectedText := "4 packets transmitted, 4 received, 0% packet loss"
+		//awsPingGcpCommand := "ping -c4 " + gcpPrivateIp
```

Comment:
Not needed? Or accidentally commented out? If not needed we should delete, or if there's a reason for this to be left here please add a comment/explanation.